cmd/cmd.go

```diff
@@ -862,16 +862,63 @@ func CopyHandler(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+type pullFn func(ctx context.Context, name string, fn api.PullProgressFunc) error
+
+func getLocalPuller(insecure bool) (p pullFn, err error) {
+	p = func(ctx context.Context, name string, fn api.PullProgressFunc) error {
+		if err := initializeKeypair(); err != nil {
+			return err
+		}
+		opts := &server.RegistryOptions{
+			Insecure: insecure,
+		}
+		f := func(r api.ProgressResponse) {
+			fn(r)
+		}
+		return server.PullModel(ctx, name, opts, f)
+	}
+	return p, nil
+}
+
+func getAPIPuller(insecure bool) (p pullFn, err error) {
+	client, err := api.ClientFromEnvironment()
+	if err != nil {
+		return p, err
+	}
+	p = func(ctx context.Context, name string, fn api.PullProgressFunc) error {
+		if err := preflightCheck(ctx, client); err != nil {
+			return err
+		}
+		req := api.PullRequest{Name: name, Insecure: insecure}
+		return client.Pull(ctx, &req, fn)
+	}
+	return p, nil
+}
+
 func PullHandler(cmd *cobra.Command, args []string) error {
 	insecure, err := cmd.Flags().GetBool("insecure")
 	if err != nil {
 		return err
 	}
+	local, err := cmd.Flags().GetBool("local")
+	if err != nil {
+		return err
+	}
 
-	client, err := api.ClientFromEnvironment()
+	var pf pullFn
+	if local {
+		fmt.Println("pulling to local machine")
+		pf, err = getLocalPuller(insecure)
+	} else {
+		fmt.Println("requesting pull from server")
```

Comment:
Thanks!

However, this isn't really logging - it's presented to the user as command line output, rather than being part of the server package.

As such, I think it's consistent with the purpose and style of the package. In the same file:

https://github.com/ollama/ollama/blob/79d3b1e2bdfc97542a7259b0c839520d39578514/cmd/cmd.go#L688
https://github.com/ollama/ollama/blob/79d3b1e2bdfc97542a7259b0c839520d39578514/cmd/cmd.go#L752
https://github.com/ollama/ollama/blob/79d3b1e2bdfc97542a7259b0c839520d39578514/cmd/cmd.go#L861
https://github.com/ollama/ollama/blob/79d3b1e2bdfc97542a7259b0c839520d39578514/cmd/cmd.go#L892
https://github.com/ollama/ollama/blob/79d3b1e2bdfc97542a7259b0c839520d39578514/cmd/cmd.go#L1049

etc. etc.